### DISCUSSION TITLE
Programmatically name pages

### DISCUSSION THREAD

**Discussion by niklasfjeldberg** [nuxt/nuxt#16310] [Questions] [answered]
How can I programmatically name pages with the composition API?

I know I can do the following with the options API:
```
export default {
  name: 'Page'
};
```
For the composition API I looked in the documentation and found:
```
definePageMeta({
  name: 'Page'
});
```
However, it does not seem to work, at least it did not work with the Chrome Extension.

**Comment by ColinEspinas** [comment, marked as answer]
Hello,
Those are actually vue features, not nuxt related. Pages are components and the `name` property is actually just the name of the component.

In vue3 there is no way to change the name from a setup function (or setup script). You should use the option API if you really need this feature. 

Here is a stack overflow explanation: https://stackoverflow.com/questions/67445608/how-to-define-name-and-inheritattrs-in-script-setup

May I ask what is the use case of your need? Nuxt should route your pages automatically from your file structure not from the components names.

**Reply by niklasfjeldberg** [reply]
I see, thank you for your answer. I use it to better debug and find components when I search in the Chrome extension. So, it's not essential.

**Comment by wobsoriano** [comment]
Try https://github.com/vbenjs/vite-plugin-vue-setup-extend
